solr/core/src/java/org/apache/solr/schema/ManagedIndexSchemaFactory.java

```diff
@@ -91,6 +92,33 @@ public void init(NamedList<?> args) {
   public String getSchemaResourceName(String cdResourceName) {
     return managedSchemaResourceName; // actually a guess; reality depends on the actual files in the config set :-(
   }
+  
+  /**
+   * Lookup the path to the managed schema, dealing with falling back to the
+   * legacy managed-schema file, instead of the expected managed-schema.xml file.
+   * 
+   * This method is duplicated in ManagedIndexSchema.
+   * @see org.apache.solr.schema.ManagedIndexSchema#lookupManagedSchemaPath
+   */
+  public String lookupManagedSchemaPath() {
+    final ZkSolrResourceLoader zkLoader = (ZkSolrResourceLoader)loader;
+    final ZkController zkController = zkLoader.getZkController();
+    final SolrZkClient zkClient = zkController.getZkClient();
+    String managedSchemaPath = zkLoader.getConfigSetZkPath() + "/" + managedSchemaResourceName;
+    try {
+      // check if we are using the legacy managed-schema file name.
+      if (!zkClient.exists(managedSchemaPath, true)){
+        log.info("Managed schema resource {} not found - loading legacy managed schema {} file instead"
+            , managedSchemaResourceName, ManagedIndexSchemaFactory.LEGACY_MANAGED_SCHEMA_RESOURCE_NAME);
+        managedSchemaPath = zkLoader.getConfigSetZkPath() + "/" + ManagedIndexSchemaFactory.LEGACY_MANAGED_SCHEMA_RESOURCE_NAME;
+      }
+    } catch (KeeperException e) {
+      throw new RuntimeException(e);
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
```

Comment:
Done....    I don't know how you knew to set that however!   Wonder if there is a check that could have flagged this?

Reply:
>  I don't know how you knew to set that however!
This is a standard Java practice that is easy to miss.  I wish Sonatype Lift / would raise alarms about this in case I'm not a code reviewer :-).  There is one bug pattern used by error-prone (Sonatype Lift calls it) -- https://errorprone.info/bugpattern/InterruptedExceptionSwallowed but you didn't _quite_ run a-fowl of that pattern since you did explicitly catch InterruptedException.  Error-prone may be assuming that because you explicitly caught it, that you know what you are doing ;-)